pkg/oauth/user.go

```diff
@@ -207,6 +210,65 @@ func (s *server) Login(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
+var (
+	errInvalidLogoutRedirectURI = errors.DefineInvalidArgument(
+		"invalid_logout_redirect_uri",
+		"the redirect URI did not match the one(s) defined in the client",
+	)
+	errMissingAccessTokenIDParam = errors.DefinePermissionDenied(
+		"missing_param_access_token_id",
+		"access token ID was not provided",
+	)
+)
+
+func (s *server) ClientLogout(c echo.Context) error {
+	ctx := c.Request().Context()
+	accessTokenID := c.QueryParam("access_token_id")
+	var redirectURI = s.config.UI.MountPath()
+	if accessTokenID == "" {
+		return errMissingAccessTokenIDParam
+	}
+	at, err := s.store.GetAccessToken(ctx, accessTokenID)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	if at != nil {
+		client, err := s.store.GetClient(ctx, &at.ClientIDs, &types.FieldMask{Paths: []string{"logout_redirect_uris"}})
+		if err != nil {
+			return err
+		}
+		redirectParam := c.QueryParam("post_logout_redirect_uri")
+		if redirectParam == "" {
+			redirectURI = client.LogoutRedirectURIs[0]
+		} else {
+			for _, uri := range client.LogoutRedirectURIs {
+				redirectURI, err = osin.ValidateUri(uri, redirectParam)
+			}
+			if err != nil {
+				return errInvalidLogoutRedirectURI.WithCause(err)
+			}
+		}
```

Comment:
Move this to the end. For existing clients in existing deployments we won't have a registered logout redirect URI, but users still need to be able to logout from those.